### ISSUE TITLE
Little more elaboration

### ISSUE THREAD

**Issue by samir-ware** [adeshpande3/LSTM-Sentiment-Analysis#7]
Hey man 
First and foremost Truly nice article and presentation . I am really new to NN and I understood what you have written in your blogpost till the point data get integerised. Although I am not able to understand what is happening after that and why. 

Also you mentioned that label placeholder represents set of values either [1,0] or [0,1] depending on weather training set is positive or negative. Although if its either positive or negative why cant we use just one dimensional array here viz either [1] or [0] ? 

May I request you to please give me some insight as to what is happening in there and how integerizing the token solves the problem ?

Thanks
Samir

**Comment by adeshpande3** [comment]
The input sequence needs to be integerized because that is what needs to be inputted into your tf.nn.embedding_lookup function needs. If you look at the [documentation](https://www.tensorflow.org/api_docs/python/tf/nn/embedding_lookup), you'll see that the function needs both an embedding matrix which contains all of the word vectors, as well as list of integerized indices that represents the sentence. Check out this [pic](https://github.com/adeshpande3/LSTM-Sentiment-Analysis/blob/master/Images/SentimentAnalysis13.png ) if you're confused 

Theoretically, you could make the output of the network as just a single scalar. However, the point I was trying to make was that this is a binary classification problem with two classes. If you wanted to extend this to multiple classes, it would be easier to make changes in the code, rather than if we just had one output value.

**Comment by samir-ware** [comment]
Thank you so much for the prompt reply. Documentation and explanation about array being two dimensional makes perfect sense. 

This may sound little stupid but can you please tell me how the output of tf.nn.embedding_lookup function solves this problem of knowing user sentiments. As far as my understanding goes , this output is the embedding matrix which is nothing but combining words with same meaning like love, adore together in the given review. Is this correct ?

I am not sure how NN get train based on the this intergerrized values returned by "tf.nn.embedding_lookup" function ? If you could explain a little bit in layman's term , it will be a great help. Any white paper or link will as well be much appreciated. 

Thanks
Samir

**Comment by adeshpande3** [comment]
Well, as I explained in the [video](https://www.oreilly.com/learning/perform-sentiment-analysis-with-lstms-using-tensorflow), the output of tf.nn.embedding_lookup gives you a sequence matrix of all the word vectors of the words in the sequence. This isn't the part that helps you with the sentiment, it's just the part that organizes the current data. 

Once you feed that data through the LSTM, that's when the network will get trained to recognize sentiment. I'd recommend looking through more links on LSTMs to learn about how they're able to store info and perform tasks like sentiment analysis. 

http://colah.github.io/posts/2015-08-Understanding-LSTMs/ is a good one

**Comment by samir-ware** [comment]
@adeshpande3 Thank you so much for the info. Appreciate it.
